sound/soc/sof/ipc4-topology.c

```diff
@@ -1540,7 +1541,16 @@ sof_ipc4_prepare_copier_module(struct snd_sof_widget *swidget,
 	ipc_config_data = &ipc4_copier->ipc_config_data;
 
 	/* config_length is DWORD based */
-	ipc_size = sizeof(*copier_data) + copier_data->gtw_cfg.config_length * 4;
+	gtw_cfg_config_length = copier_data->gtw_cfg.config_length * 4;
+	ipc_size = sizeof(*copier_data) + gtw_cfg_config_length;
+
+	if (ipc4_copier->dma_config_tlv.type == SOF_IPC4_GTW_DMA_CONFIG_ID &&
+	    ipc4_copier->dma_config_tlv.length) {
+		ipc_size += sizeof(ipc4_copier->dma_config_tlv);
```

Comment:
this will exclude the size of the `ipc4_copier->dma_config_tlv.dma_config.dma_priv_config` ?

There is a mismatch here, I think. The type is not 0 (it is 0x1000) and we check if `ipc4_copier->dma_config_tlv.length` is not 0, which cannot happen as per the comment in the previous patch it should be set to some value (`sizeof(struct sof_ipc4_dma_config)`,  which I think is not correct), then we discard that length information and do another sizeof on the struct? But the `sizeof(ipc4_copier->dma_config_tlv)` will return `sizeof(struct sof_ipc4_dma_config)` and that sizeof _excludes the size of the `dma_priv_config`.

Or am I missing something?

Reply:
fixed